pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -93,11 +92,43 @@ contract IndexPool is BaseWeightedPool {
         // Immutable variables cannot be initialized inside an if statement, so we must do conditional assignments
         _tokens = tokens;
 
-        for(uint i = 0; i < numTokens; i++){
+        for (uint8 i = 0; i < numTokens; i++) {
             scalingFactors.push(_computeScalingFactor(tokens[i]));
         }
     }
 
+    function reweighTokens(address[] memory tokens, uint96[] memory normalizedWeights) public {
```

Comment:
In the requirement doc, you called the variable `desiredWeights`. I thought that is a good name for it because it's a future state. Can we change this name to that as well? Same for the other function I think

Reply:
Also you used `calldata` in the doc. Seeing that we don't do any logic with the parameters and `calldata` is cheaper, why did you choose to go for `memory`? Just curious